tldw_Server_API/app/core/AuthNZ/database.py

```
@@ -123,9 +147,37 @@ def _should_use_postgres(self) -> bool:
-        """Resolve sqlite connection string, uri flag, and filesystem path."""
+        """Resolve sqlite connection string, uri flag, and filesystem path.
+
+        Defensive behavior: when running in single-user mode and provided URL
+        is not a sqlite/file scheme, fall back to the default SQLite users DB
+        path instead of treating the URL as a filesystem path.
+        """
+        # Determine auth mode to decide fallback handling for non-sqlite schemes
+        try:
+            from tldw_Server_API.app.core.AuthNZ.settings import get_settings as _get_settings  # local import to avoid cycles
+            _auth_mode = str(getattr(_get_settings(), "AUTH_MODE", "single_user")).strip().lower()
+        except Exception:
```

Ruff (0.14.4)

Do not catch blind exception: `Exception`

(BLE001)